arbitrator/include/capabilities_interface.tpp

```diff
@@ -23,38 +23,87 @@
+            return responses;
+        }
+
+        retry_attempt = 0;
+        while (!topics_to_retry.empty() && retry_attempt < MAX_RETRY_ATTEMPTS)
+        {
+            current_topics_to_check = topics_to_retry;
+            topics_to_retry.clear();
+            for (const auto & topic : current_topics_to_check)
+            {
+                try {
+                    using std::literals::chrono_literals::operator""ms;
 
-                const auto client = nh_->create_client<carma_planning_msgs::srv::PlanManeuvers>(topic);
-                RCLCPP_DEBUG_STREAM(rclcpp::get_logger("arbitrator"), "found client: " << topic);
+                    const auto client = nh_->create_client<carma_planning_msgs::srv::PlanManeuvers>(topic);
+                    RCLCPP_DEBUG_STREAM(rclcpp::get_logger("arbitrator"), "found client: " << topic);
 
-                const auto response = client->async_send_request(msg);
+                    const auto response = client->async_send_request(msg);
 
-                switch (const auto status{response.wait_for(500ms)}; status) {
-                    case std::future_status::ready:
-                        responses.emplace(topic, response.get());
-                        break;
-                    case std::future_status::deferred:
-                    case std::future_status::timeout:
-                        RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"), "failed...: " << topic);
+                    switch (const auto status{response.wait_for(500ms)}) {
+                        case std::future_status::ready:
+                            responses.emplace(topic, response.get());
+                            break;
+                        case std::future_status::deferred:
+                        case std::future_status::timeout:
+                            RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"), "failed...: " << topic);
+                    }
+                    if (retry_attempt > 0)
+                    {
+                        RCLCPP_INFO_STREAM(rclcpp::get_logger("arbitrator"),"Service call to: " << topic << ", successfully finished after retrying: " << retry_attempt);
+                    }
+
+                } catch(const rclcpp::exceptions::RCLError& error) {
+                    RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"),
+                        "Cannot make service request for service '" << topic << "': " << error.what() <<". So retrying, attempt no: " << retry_attempt);
+                    topics_to_retry.push_back(topic);
+                    continue;
                 }
-            } catch(const rclcpp::exceptions::RCLError& error) {
-                RCLCPP_ERROR_STREAM(rclcpp::get_logger("arbitrator"),
-                    "Cannot make service request for service '" << topic << "': " << error.what());
-                continue;
             }
+            retry_attempt ++;
+        }
+
+        if (retry_attempt == MAX_RETRY_ATTEMPTS)
```

Comment:
Just for safety we should check for retry_attempt >= MAX_RETRY_ATTEMPT incase we have a logically error some were